### PR TITLE
feat(api): 2158 - use common young data to generate convocation

### PR DESCRIPTION
**Pull request by Johannbr**
**Pull Request Title Rules**

Pull request format must respect this format:

```
<type>(<scope>, <scope>...): <Notion ID> - <description>
```

Allowed types: `feat`, `fix`, `refactor`, `chore`.
Allowed scopes: `release`, `app`, `admin`, `misc`, `api`, `all`, `github`, `terraform`, `kb`, `lib`.

Notion ID, followed by a `-` is **mandatory** on types `refactor`, `feat` and `fix`.

*If you really need to bypass the pull request title validation, you can add the `bypass title` label to you PR.*

Examples of valid PR titles:
- `chore(release): New version 1.3.3.7`
- `refactor(api): 1789 - Removed monarchy from codebase`
- `fix(api, app, admin): 1981 - User cannot delete each other accounts anymore.`

**Description**

**Todo**

**Checklist**

- [ ] **⚠️ My code can have side-effects on other part of the code-base**
- [ ] I have added the Plausible tags/events
- [ ] I have performed a self-review of my code (and removed console.log)

**Ticket / Issue**

**Testing instructions**

### CODE REVIEW

File: api/src/templates/convocation/cohesion.js [hunk 1]
```diff
@@ -255,19 +254,39 @@ async function generateCohesion(outStream, young) {
...
+  const validatedYoungs = youngs.filter((young) => young.status === YOUNG_STATUS.VALIDATED);
   const doc = initDocument({ autoFirstPage: false });
   doc.pipe(outStream);
-  for (const young of youngs) {
+  let commonYoungData = await getYoungCommonData(validatedYoungs);
+
+  for (const young of validatedYoungs) {
     try {
-      const data = await fetchDataForYoung(young);
       doc.addPage();
```
Comment (achorein) [inline api/src/templates/convocation/cohesion.js]: il vaudrait mieux ajouter la page après le contrôle de données, sinon en cas d'erreur on se retrouvera avec des pages blanches.

File: api/src/classe/classe.service.test.ts [hunk 2]
```diff
@@ -6,20 +6,6 @@ const findYoungsByClasseIdSpy = jest.spyOn(youngService, "findYoungsByClasseId")
...
 
 describe("ClasseService", () => {
-  it("should throw an error", async () => {
-    findYoungsByClasseIdSpy.mockReturnValue(Promise.resolve(new Array(51).fill({})));
-
-    try {
-      const resultPdf = await classService.generateConvocationsByClasseId("classeId");
-      expect(findYoungsByClasseIdSpy).toHaveBeenCalledTimes(1);
-
-      await expect(resultPdf).rejects.toThrow(FUNCTIONAL_ERRORS.TOO_MANY_YOUNGS_IN_CLASSE);
```
Comment (achorein) [inline api/src/classe/classe.service.test.ts:16]: l'import de `FUNCTIONAL_ERRORS` n'est plus nécessaire

**Review by achorein — APPROVED** [review]
LGTM

File: api/src/classe/classe.service.test.ts (hunk 2 above)
Comment (Johannbr) [reply api/src/classe/classe.service.test.ts:16]: done

File: api/src/templates/convocation/cohesion.js (hunk 1 above)
Comment (Johannbr) [reply api/src/templates/convocation/cohesion.js]: done

**Pull request merged**